compiler/backend_inkwell/src/lib.rs

```diff
@@ -0,0 +1,1041 @@
+                candy_type_ptr.into(),
+                i64_type.into(),
+            ],
+            false,
+        );
+        self.module.add_function(
+            "make_candy_function",
+            make_function_fn_type,
+            Some(Linkage::External),
+        );
+
+        let make_struct_fn_type =
+            candy_type_ptr.fn_type(&[candy_type_ptr.into(), candy_type_ptr.into()], false);
+        self.module.add_function(
+            "make_candy_struct",
+            make_struct_fn_type,
+            Some(Linkage::External),
+        );
+
+        let struct_get_fn_type =
+            candy_type_ptr.fn_type(&[candy_type_ptr.into(), candy_type_ptr.into()], false);
+        let candy_builtin_struct_get =
+            self.module
+                .add_function("candy_builtin_struct_get", struct_get_fn_type, None);
+
+        let panic_fn_type = void_type.fn_type(
+            &[candy_type.ptr_type(AddressSpace::default()).into()],
+            false,
+        );
+        self.module.add_function("candy_panic", panic_fn_type, None);
+        let free_fn = self
+            .module
+            .add_function("free_candy_value", panic_fn_type, None);
+        let print_fn = self
+            .module
+            .add_function("print_candy_value", panic_fn_type, None);
+
+        let candy_fn_type = candy_type_ptr.fn_type(&[], true);
+        let get_candy_fn_ptr_type = candy_fn_type
+            .ptr_type(AddressSpace::default())
+            .fn_type(&[candy_type_ptr.into()], false);
+        self.module
+            .add_function("get_candy_function_pointer", get_candy_fn_ptr_type, None);
+        let get_candy_fn_env_type = candy_type_ptr.fn_type(&[candy_type_ptr.into()], false);
+        self.module.add_function(
+            "get_candy_function_environment",
+            get_candy_fn_env_type,
+            None,
+        );
+
+        let main_type = i32_type.fn_type(&[], false);
+        let main_fn = self.module.add_function("main", main_type, None);
+        let block = self.context.append_basic_block(main_fn, "entry");
+
+        let call_candy_function_type =
+            candy_type_ptr.fn_type(&[candy_type_ptr.into(), candy_type_ptr.into()], false);
+        let call_candy_function_with =
+            self.module
+                .add_function("call_candy_function_with", call_candy_function_type, None);
+
+        let main_info = FunctionInfo {
+            function_value: Rc::new(main_fn),
+            captured_ids: vec![],
+            env_type: None,
+        };
+
+        self.builder.position_at_end(block);
+        self.compile_mir(&self.mir.body.clone(), &main_info);
+        self.builder.position_at_end(block);
+
+        let environment = self
+            .module
+            .add_global(candy_type_ptr, None, "candy_environment");
+
+        if let Some(main_return) = self.main_return {
+            const MAIN_FN_NAME: &str = "Main";
+            let len = i32_type.const_int(MAIN_FN_NAME.len() as u64, false);
```

Comment:
Shouldn't this be one longer byte to account for the null terminator?

```suggestion
            let len = i32_type.const_int(MAIN_FN_NAME.len() as u64 + 1, false);
```